### ISSUE TITLE
Basic Node Telemetry

### ISSUE THREAD

**Issue by adamkrellenstein** [CounterpartyXCP/counterparty-core#1522]
So Counterparty nodes don't talk to each other directly. This means that there's currently no way of knowing even the most basic telemetry for who's running what version, etc. which is important for managing things like upgrades and hotfixes. Bitcoin itself handles this through its peer-to-peer network.[[1](https://en.bitcoin.it/wiki/Protocol_documentation#Message_types)] Obviously all of this data is 100% anonymous and there'll be a flag to disable it entirely. Will be shared on a public dashboard.

* Version / DB Version
* Uptime
* Platform (inc. Docker vs. `simplenode` vs `fednode`)
* Network (`mainnet`/`testnet`/etc.)
* Last block parsed
* `--force` enabled
* Block, ledger and message hashes

**Comment by mattmarcello** [comment]
Could you clarify what is meant by:

- `--force` enabled

**Comment by adamkrellenstein** [comment]
Sorry, what I meant was that nodes should report if they're running with the `--force` CLI flag enabled (which would indicate that certain safety checks are being bypassed, etc.)

**Comment by mattmarcello** [comment]
Understood

**Comment by mattmarcello** [comment]
@adamkrellenstein just to be clear, are you cool with a handler at `/telemetry` from which a future dashboard can pull?  Or do you want it to push data to some telemetry backend at regular intervals?

**Comment by adamkrellenstein** [comment]
I think we need to set up the push.

**Comment by adamkrellenstein** [comment]
I think we'll also want to expose all of this info at some REST endpoint, however, and merge it with the `get_running_info` we have now:

```
adam@tang ~ > http POST https://rpc:rpc@api0.counterparty.io:4000/api/ jsonrpc="2.0" id=1 method="get_running_info"
HTTP/1.1 200 OK
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Content-Length: 890
access-control-allow-headers: DNT,X-Mx-ReqToken,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Authorization
access-control-allow-methods: GET, POST, OPTIONS
access-control-allow-origin: *
content-type: application/json
date: Wed, 03 Apr 2024 19:41:07 GMT
server: Werkzeug/3.0.1 Python/3.10.12
via: 1.1 google

{
    "id": "1",
    "jsonrpc": "2.0",
    "result": {
        "api_limit_rows": 1000,
        "bitcoin_block_count": 837583,
        "db_caught_up": true,
        "indexd_blocks_behind": 0,
        "indexd_caught_up": true,
        "last_block": {
            "block_hash": "00000000000000000002940a1691b138b690f23a86d8f32decdca46dd62e76c1",
            "block_index": 837583,
            "block_time": 1712173088,
            "difficulty": 83126997340024.61,
            "ledger_hash": "e3f9cd46f2b2944903800d7257a92842497d3cf0a28a156fe2ca302cf15c928f",
            "messages_hash": "8bfd8a1d33fb3909498575e4721672c314215b5d0a85d1041c88c593e7841d94",
            "previous_block_hash": "00000000000000000002a0f8fffb94bf8e48816190cf99bfa1f7c97b8b690eaa",
            "txlist_hash": "2295a65d8e5f0efc10095e1f9e611fd3e9c627bce0216c61f8f5d89ae7d15040"
        },
        "last_message_index": 13611493,
        "running_regtest": false,
        "running_testcoin": false,
        "running_testnet": false,
        "server_ready": true,
        "version_major": 10,
        "version_minor": 0,
        "version_revision": 0
    }
}
```

**Comment by adamkrellenstein** [comment]
And we should also expose this info for Sentry.

**Comment by mattmarcello** [comment]
Okay, here is data I am collecting: 

```
 {
      "version": "10.1.0",
      "addrindex_version": "0.4.6",
      "uptime": "472.56",
      "is_docker_process": false,
      "network": "MAINNET",
      "force_enabled": false,
      "last_block": {
        "block_index": 839529,
        "block_hash": "0000000000000000000330586ef1577def4246f5900bd8b6bc01ce562172b258",
        "block_time": 1713294948,
        "previous_block_hash": "000000000000000000034158a91c1876f5fc2add1e69641e908956ac9de45b93",
        "difficulty": 86388558925171.02,
        "ledger_hash": "23ccfaa264719b760f4950d0ec7c07641129f4d43a979eeaa8fdd7950be336a9",
        "txlist_hash": "557213d08a66c89d84475387deca135bd164075b6e663741a917ca36f78630b6",
        "messages_hash": "fb8f1f4225a9aba71375d4c402d88bddf8390672735819d735847892995abe93"
      }
    }

```

In terms of merging with `get_running_info`, a lot of the fields are duplicative even if they are encoded slightly differently ( e.g. network="MAINNET" vs is_testnet=false, or encoding version as "version_major", "version_minor", "version-revision" vs just a single string. 

Here is one possible shape that is non-breaking:

```

{
    "server_ready": true,
    "db_caught_up": true,
    "bitcoin_block_count": 839529,
    "last_block": {
      "block_index": 839529,
      "block_hash": "0000000000000000000330586ef1577def4246f5900bd8b6bc01ce562172b258",
      "block_time": 1713294948,
      "previous_block_hash": "000000000000000000034158a91c1876f5fc2add1e69641e908956ac9de45b93",
      "difficulty": 86388558925171.02,
      "ledger_hash": "23ccfaa264719b760f4950d0ec7c07641129f4d43a979eeaa8fdd7950be336a9",
      "txlist_hash": "557213d08a66c89d84475387deca135bd164075b6e663741a917ca36f78630b6",
      "messages_hash": "fb8f1f4225a9aba71375d4c402d88bddf8390672735819d735847892995abe93"
    },
    "indexd_caught_up": true,
    "indexd_blocks_behind": 0,
    "last_message_index": 13650823,
    "api_limit_rows": 1000,
    "running_testnet": false,
    "running_regtest": false,
    "running_testcoin": false,
    "version_major": 10,
    "version_minor": 1,
    "version_revision": 0,
    # new fields  
    "addrindexrs_version_major": 10,
    "addrindexrs_version_minor": 1,
    "addrindexrs_version_revision": 0, 
    "uptime": "472.56",  
    "is_docker_process": false, 
     "force_enabled": false,
}
```

**Comment by adamkrellenstein** [comment]
I think we should just take the opportunity to deprecate `get_running_info` as part of the API Redesign and keep it how it is but put it behind `/api/v1/`

**Comment by adamkrellenstein** [comment]
thoughts @ouziel-slama ? anything we're missing?

**Comment by ouziel-slama** [comment]
yeah I agree! Apart from the bug fix I would say that we freeze the RPC API and focus on the v2 API.
Could we add `mempool-enabled` and `checkdb-enabled`?
I will add `api-v1-enabled` in the `newapi` branch!

**Comment by adamkrellenstein** [comment]
I think let's leave out `checkdb-enabled`, since it's really just a startup thing.

**Comment by adamkrellenstein** [comment]
Some of this was included in v10.1.1... how much? @mattmarcello

**Comment by mattmarcello** [comment]
All it does is collect the telemetry attributes at a set interval and log them to local storage.  If you have sentry enabled it will also send some of those attributes to sentry when there is an exception ( e.g adressindexrs version )
